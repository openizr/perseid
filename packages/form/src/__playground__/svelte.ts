/* c8 ignore start */
/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call */

import { SvelteComponent } from 'svelte';
import Engine from 'scripts/core/Engine';
import Form from 'scripts/svelte/Form.svelte';
import Field from '__playground__/Field.svelte';

let app: Form;
const { log } = console;

function main(): void {
  app = new Form({
    props: {
      Field: Field as typeof SvelteComponent | null | undefined,
      configuration: {
        root: 'root',
        fields: {
          test: { type: 'string', required: true },
        },
        steps: {
          root: {
            fields: ['test'],
          },
        },
        plugins: [
          (engine: Engine): void => {
            engine.on('userAction', (data, next) => {
              log(data);
              return next(data);
            });
          },
        ],
      },
    },
    target: document.getElementById('root') as unknown as HTMLElement,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { $destroy: () => void; }).$destroy();
});
