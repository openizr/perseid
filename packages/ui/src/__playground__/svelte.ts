/* c8 ignore start */

import type { SvelteComponent } from 'svelte';
import Router from '__playground__/pages/Router.svelte';

let app: SvelteComponent;

function main(): void {
  const target = document.querySelector('#root') as unknown as HTMLElement;
  target.innerHTML = '';
  app = new Router({
    hydrate: false,
    target,
  });
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Svelte components tree.
window.addEventListener('beforeunload', () => {
  app.$destroy();
});
