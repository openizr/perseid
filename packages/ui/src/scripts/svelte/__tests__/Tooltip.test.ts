/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render, fireEvent } from '@testing-library/svelte';
import UITooltip from 'scripts/svelte/Tooltip.svelte';
import TestUITooltip from 'scripts/svelte/__tests__/TestTooltip.svelte';
import TestUITooltip2 from 'scripts/svelte/__tests__/TestTooltip2.svelte';

describe('svelte/UITooltip', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const { container } = render(UITooltip, { props: { label: 'Test', modifiers: 'top' } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - basic 2', async () => {
    const { container } = render(TestUITooltip2);
    const button = container.getElementsByTagName('button')[0];
    await fireEvent.keyPress(button);
    expect(container.firstChild).toMatchSnapshot();
    await fireEvent.focusOut(button);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with description', async () => {
    const { container } = render(TestUITooltip);
    const button = container.getElementsByTagName('button')[0];
    await fireEvent.keyPress(button);
    expect(container.firstChild).toMatchSnapshot();
    await fireEvent.focusOut(button);
    expect(container.firstChild).toMatchSnapshot();
  });
});
