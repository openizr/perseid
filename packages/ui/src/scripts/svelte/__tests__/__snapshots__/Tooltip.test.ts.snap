// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte/UITooltip > renders correctly - basic 1`] = `
<div>
  <div
    aria-label="Test"
    class="ui-tooltip ui-tooltip--top"
    role="tooltip"
  >
     
  </div>
  <!--&lt;Tooltip&gt;-->
</div>
`;

exports[`svelte/UITooltip > renders correctly - basic 2 1`] = `
<div>
  <div
    aria-label="Test"
    class="ui-tooltip"
    role="tooltip"
  >
    <button>
      i
    </button>
     
  </div>
  <!--&lt;Tooltip&gt;-->
  <!--&lt;TestTooltip2&gt;-->
</div>
`;

exports[`svelte/UITooltip > renders correctly - basic 2 2`] = `
<div>
  <div
    aria-label="Test"
    class="ui-tooltip"
    role="tooltip"
  >
    <button>
      i
    </button>
     
  </div>
  <!--&lt;Tooltip&gt;-->
  <!--&lt;TestTooltip2&gt;-->
</div>
`;

exports[`svelte/UITooltip > renders correctly - with description 1`] = `
<div>
  <div
    aria-label="Test"
    class="ui-tooltip ui-tooltip--top"
    role="tooltip"
  >
    <button>
      i
    </button>
     
  </div>
  <!--&lt;Tooltip&gt;-->
  <!--&lt;TestTooltip&gt;-->
</div>
`;

exports[`svelte/UITooltip > renders correctly - with description 2`] = `
<div>
  <div
    aria-label="Test"
    class="ui-tooltip ui-tooltip--top"
    role="tooltip"
  >
    <button>
      i
    </button>
     
  </div>
  <!--&lt;Tooltip&gt;-->
  <!--&lt;TestTooltip&gt;-->
</div>
`;
